### ISSUE TITLE
Unique Settings for Different Programming Environments

### ISSUE THREAD

**Issue by MRItopMD** [microsoft/vscode#98093]
## Motivation
VS Code is a fantastic editor, in that, it allows you to often balance IDE-like features with Editor-like features via the base platform + extensions. This, of course, means if I like to program in multiple languages, I don't have to jump from IDE to IDE and the resource intensiveness that follows from that. This, however, could be made more efficient. VS Code is already fairly fast to startup, but the more extensions you have, understandably the worst the performance can get. Large projects, of course, can make the software slow down in certain situations. So, ideally, we want to cut down on bloat where necessary. This is where my multiple settings environment idea comes from

## Concept
I suggest the creation of "profiles" which will allow extension enablement, settings(such as typeface, themes etc.) to be made specific to each profile. Such as if I select "Profile: Python" it would use a pre-populated Python_settings.json to activate my preferred extensions(not sure if that's the right or correct design/programming decision, just spitballing), disable ones I wouldn't use(such as if I'm solely coding python, I may not want to have my C++ and LaTeX extensions running, but still want my markdown extension running for my docs). I could even make preferences such as different text sizes, different color themes, different preferred terminal, for different profiles. Or just to give more examples, sometimes I use python for quick scripting, other times I'm doing intensive data science with various visualizations, extensions for CSV, LaTeX, Jupyter etc. And in that context, I may also use other languages like R or such if I'm trying to load code sent by a colleague. So in that context, I may want to package the extensions I want to use based on the use case, such Profile: Data Science. 

## Conclusion

In Conclusion, I believe this "profiles" idea will vastly improve the versatility of the platform for those who use VS Code for multiple types of development and/or languages. In addition, it should improve performance for those who want to have. And I do think because each profile will deactivate unused extensions for that profile, it will allow developers to create more comprehensive development environments with extensions without fear of having too many that would slow down performance. With profiles, a much more complete experience can be created.

**Comment by sandy081** [comment]
/duplicate

There is already a feature request to support profiles.

**Comment by wolf99** [comment]
See #92393
